es/normalizeOptions.js

```diff
@@ -13,6 +13,19 @@ var toArray = function toArray(o) {
   return o ? [].concat(o) : [];
 };
 
+var addressStack = [];
+var defaultOnRenderStart = function defaultOnRenderStart(displayName) {
+  addressStack.push(displayName);
+  console.log();
+};
+var defaultOnRenderEnd = function defaultOnRenderEnd(displayName, diff) {
+  if (addressStack[addressStack.length - 1] !== displayName) {
+    throw new Error('expecting ' + displayName + ' as top-most on addressStack, instead got ' + addressStack[addressStack.length - 1]);
```

Comment:
@AshleyColemanBigHealth why do you need to throw the errors?

Reply:
Hm. Well because I didn't understand what to expect from React & the reconciliation process now and in the future.

If the question is, _Why not allow the default reference error?_ 
Just to provide more context to future me.